webpack/scenes/ContentViews/Publish/CVPublishReview.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+import { translate as __ } from 'foremanReact/common/I18n';
+import ContentViewIcon from '../components/ContentViewIcon';
+import InactiveText from '../components/InactiveText';
+import ComponentEnvironments from '../Details/ComponentContentViews/ComponentEnvironments';
+import { selectEnvironmentPaths } from '../components/EnvironmentPaths/EnvironmentPathSelectors';
+
+const CVPublishReview = ({
+  details,
+  userCheckedItems,
+}) => {
+  const environmentPathResponse = useSelector(selectEnvironmentPaths);
+  const { results } = environmentPathResponse || {};
+  const environmentsAdded = (environmentResult) => {
+    let environmentsList = [];
+    let library = null;
+    environmentResult.forEach((path) => {
+      const {
+        environments,
+      } = path || {};
+      [library] = environments;
+      const envInPath = environments.filter(env =>
+        userCheckedItems.filter(item => item.id === env.id).includes(env));
```

Comment:
filter-inception!  Can this be clarified with a comment or a named function?

Reply:
Simplified this whole thing a little and used useMemo this time. This adds library to the userCheckedItems to display in the review table. 

```
const promotedToEnvironments = useMemo(() => {
    if (!environmentPathLoading) {
      const { results } = environmentPathResponse || {};
      // 1st env in result's every path is Library
      return [results[0].environments[0]].concat(userCheckedItems);
    }
    return [];
  }, [environmentPathResponse, environmentPathLoading]);
```